azure-iot-device/azure/iot/device/iothub/transport/mqtt/mqtt_transport.py

```diff
@@ -329,84 +323,32 @@ def _on_provider_disconnect_complete(self):
             self._disconnect_callback = None
             callback()
 
-    def _on_provider_publish_complete(self, mid):
-        """
-        Callback that is called by the provider when it receives a PUBACK from the service
-
-        :param mid: message id that was returned by the provider when `publish` was called.  This is used to tie the
-            PUBLISH to the PUBACK.
-        """
-        if mid in self._in_progress_actions:
-            callback = self._in_progress_actions[mid]
-            del self._in_progress_actions[mid]
-            callback()
-        else:
-            logger.warning("PUBACK received with unknown MID: %s", str(mid))
-            self._responses_with_unknown_mid[
-                mid
-            ] = mid  # storing MID for now.  will probably store result code later.
-
-    def _on_provider_subscribe_complete(self, mid):
-        """
-        Callback that is called by the provider when it receives a SUBACK from the service
-
-        :param mid: message id that was returned by the provider when `subscribe` was called.  This is used to tie the
-            SUBSCRIBE to the SUBACK.
-        """
-        if mid in self._in_progress_actions:
-            callback = self._in_progress_actions[mid]
-            del self._in_progress_actions[mid]
-            callback()
-        else:
-            logger.warning("SUBACK received with unknown MID: %s", str(mid))
-            self._responses_with_unknown_mid[
-                mid
-            ] = mid  # storing MID for now.  will probably store result code later.
-
     def _on_provider_message_received_callback(self, topic, payload):
         """
         Callback that is called by the provider when a message is received.  This message can be any MQTT message,
         including, but not limited to, a C2D message, an input message, a TWIN patch, a twin response (/res), and
         a method invocation.  This function needs to decide what kind of message it is based on the topic name and
         take the correct action.
 
-        :param topic: MQTT topic name that the message arrived on
-        :param payload: Payload of the message
+        :param str topic: MQTT topic name that the message arrived on
+        :param bytes payload: Payload of the message as a bytestring
         """
-        logger.info("Message received on topic %s", topic)
+        logger.info("Message received on topic {}".format(topic))
```

Comment:
>logger.info("Message received on topic {}".format(topic)) [](start = 8, length = 57)

I'm curious why you changed this.  Is this personal preference or do you have a specific reason?

Reply:
.format is considered to be the more modern way to do it. It's more flexible and avoids a lot of problems related to the %s approach. In this context, there's no real difference, it was more of a consistency thing. Since I learned the convention is now to use format, I've been trying to switch over to that exclusively.

---
In reply to: [277841533](https://github.com/Azure/azure-iot-sdk-python-preview/pull/82#discussion_r277841533) [](ancestors = 277841533)